io/io/inc/TFileBufferRead.h

```diff
@@ -0,0 +1,67 @@
+#ifndef ROOT_TFileBufferRead
+#define ROOT_TFileBufferRead
+
+/**
+ * Add a buffering layer in front of a TFile.
+ *
+ * This divides the file into 128MB chunks; the first time a byte from a chunk
+ * is accessed, the whole chunk is downloaded to a temporary file on disk.  A
+ * file chunk is only downloaded if accessed.
+ *
+ * This is *not* meant to be a cache: as soon as the TFile is closed, the buffer
+ * is removed.  This means repeated access to the same file will still cause it
+ * to be downloaded multiple times.
+ *
+ * Rather, the focus of this buffer is to hide the effects of network latency.
+ *
+ * Implementation is based on a similar approach from CMSSW:
+ * https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/Utilities/StorageFactory/src/LocalCacheFile.cc
+ */
+class TFile;
+
+class TFileBufferRead {
+public:
+  TFileBufferRead(TFile *file);
+  ~TFileBufferRead();
+
+  /**
+   * Positional read of the file, using the underlying buffer file.
+   *
+   * If the buffer file is invalid, then this reads from the underlying file.
+   *
+   * On error, -1 is returned and errno is set appropriately.
+   */
+  ssize_t pread(char *into, size_t n, off_t pos);
```

Comment:
Could you update the file for naming convention (camel case for funciton name) spacing (indention 3 spaces)?

Thanks.